### PR TITLE
🗑️  Delete submission endpoint

### PR DESCRIPTION
**Pull request by sdevalapurkar-bcgov**
# Description

This PR includes the following proposed change(s):

- Delete submission endpoint (backend and frontend changes)
- refactor demo app to use react dropzone to be consistent with frontend
- add better tests for demo app document upload, generate url and efile
- add loader to efile my package button

- JIRA: https://justice.gov.bc.ca/jira/browse/FLA-285

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- local
- jest
- storybook

### CODE REVIEW

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/document/DocumentStoreImplTest.java [hunk 1]
```diff
@@ -85,4 +84,4 @@ public void withCourtLevelCourtClassShouldReturnDocumentTypes() {
 
     }
 
-}
+}
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/document/DocumentStoreImplTest.java:87]: nit: missing newline

Comment (akroon3r) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/document/DocumentStoreImplTest.java:87]:
```suggestion
}

```

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/document/DocumentStoreImpl.java [hunk 2]
```diff
@@ -28,6 +29,12 @@ public DocumentStoreImpl(EfilingDocumentService efilingDocumentService) {
         return null;
     }
 
+    @Override
+    @Cacheable(cacheNames = "document", key = "#compositeId", cacheManager = "documentCacheManager")
+    public void evict(String compositeId) {
+        //This implements Redis delete no code required
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/document/DocumentStoreImpl.java:35]: awesome comment

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/document/DocumentStoreImplTest.java [hunk 3]
```diff
@@ -59,6 +60,12 @@ public void withoutCacheShouldReturnNull() {
         Assertions.assertNull(sut.get("id"));
     }
 
+    @Test
+    @DisplayName("OK: evict should do a thing")
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/document/DocumentStoreImplTest.java:64]: can this display name be something that describes what it does like:
```OK: evict should delete submission```

Comment (alexjoybc) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/document/DocumentStoreImplTest.java:64]:
```suggestion
    @DisplayName("OK: evict should delete submission")
```

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java [hunk 4]
```diff
@@ -300,10 +301,30 @@ private UserDetails buildUserDetails(UUID universalId) {
...
 
 
+    @Override
+    public ResponseEntity<Void> deleteSubmission(UUID submissionId, UUID xTransactionId) {
+
+        Optional<Submission> fromCacheSubmission = this.submissionStore.get(submissionId, xTransactionId);
+        if(!fromCacheSubmission.isPresent())
+            return ResponseEntity.notFound().build();
+        //Remove documents from cache
+        fromCacheSubmission.get().getFilingPackage().getDocuments().forEach(
```
Comment (alexjoybc) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:311]: we should make that one null safe

**Pull request merged**